webapp/src/app.tsx

```diff
@@ -67,7 +67,16 @@ interface IAppState {
     showParts?: boolean;
 }
 
+interface ExternalMessageData {
+    messageType: string;
+    args: any
+}
+interface ExternalMessageResult {
+    error?: any;
+    result?: any;
+}
 
+let isElectron = /[?&]electron=1/.test(window.location.href);
```

Comment:
The comment concerning the usage of flags that include the name of a specific implementation of non-browser solution of presenting pxt(-microbit).

- It is conceivable that names of specific implementations can change leaving historical names in the code: been there, refactored it.
- It causes a specific implementation focus in code that should not be concerned with what is using it's services: one begins to forget that other Open Source users of the code would like to avoid explaining that it is not, for example, electron that is being used.

Please consider using  a type, example nonBrowser, as the facility can be useful to other implementations, be they electron or other specific implementations of encapsulating the pxt(-microbit)